src/commonMain/kotlin/com/github/nanoflakes/Nanoflake.kt

```diff
@@ -3,57 +3,39 @@ package com.github.nanoflakes
 /**
  * A [Nanoflake](https://github.com/nanoflakes/nanoflakes).
  */
-class Nanoflake {
-    /**
-     * The base epoch of the nanoflake.
-     */
-    private val epoch: Long
-
+public sealed interface Nanoflake {
```

Comment:
I don't see a point in sealing the interface, TBH

IMO, keeping it open would be best for everyone (specially people implementing libraries atop nanoflakes-kotlin)